### PR TITLE
test: increase coverage to 99.3% with 83 new unit tests

### LINKED ISSUE #65: Ensure 99.5% code coverage and E2E tests cover all features
## Goal
- Raise unit test code coverage threshold from 98% to 99.5%
- Ensure E2E tests cover all app features

## Scope
- Identify uncovered branches/methods and add unit tests to reach 99.5% line/branch coverage
- Review existing E2E shards and add missing feature coverage (repeat/schedule tasks, resume interrupted Pomodoro, auto-start session, etc.)
- Update CI coverage threshold in codecov.yml

### PR DESCRIPTION
**Pull request by Nayeem170**
## Summary
- Added 83 new unit tests covering TaskEditPanel, TaskItemComponent, TaskList, TimerDisplay, WeeklyMiniChart, Index page, History page, and TaskService
- Coverage improved from 97% to 99.3% (6078/6119 lines covered)
- Updated codecov threshold from 98% to 99%

## Coverage Details
- TaskEditPanel: 0% to 95.4% (ToggleWeekday, HandleSave, OnInitialized tests)
- TaskItemComponent: 97.9% to 100% (tooltip, edit badge tests)
- TaskList: 94.9% to 98.5% (HandleTaskEdit callback test)
- TimerDisplay: 98.4% to 100% (CurrentIsRunning property test)
- WeeklyMiniChart: 98% to 100% (DictionariesEqual, parameter change skip tests)
- Index page: 58% to 60% (exception paths, success paths, cache behavior)
- History page: 82% to 82.1% (UpdateFormattedDate tests)
- TaskService: 99% to 99.5% (endDate boundary test)

## Remaining Uncovered Lines (41)
- Program.cs entry point (5 lines) - cannot unit test
- CloudSyncService Task.Run/timer callbacks (14 lines) - async fire-and-forget patterns
- TaskService/ImportService dead code (5 lines) - unreachable switch defaults
- Razor template lines (13 lines) - loading states, conditional rendering
- Index code-behind closing braces (4 lines) - coverage tool branch tracking

Closes #65

## Summary by CodeRabbit

* **Tests**
  * Expanded test coverage across UI and logic: task editing flow, list interactions, repeat/schedule badges, recurring-task rules and edge cases, history/timer behaviors, chart re-render stability, branch/error paths, and numerous component/unit scenarios.
* **Chores**
  * Increased code coverage targets from 98% to 99.5%.

### CODE REVIEW

**Review by Nayeem170 — COMMENTED** [review]
All CodeRabbit feedback resolved in PR #67.

File: tests/Pomodoro.Web.Tests/Pages/IndexCoverageTests.cs [hunk 1]
```diff
@@ -0,0 +1,500 @@
...
+
+    [Fact]
+    public async Task HandleTimerStart_PomodoroWithTask_StartsPomodoro()
+    {
+        var taskId = Guid.NewGuid();
+        TaskServiceMock.SetupGet(x => x.CurrentTaskId).Returns(taskId);
+        TimerServiceMock.Setup(x => x.StartPomodoroAsync(taskId)).Returns(Task.CompletedTask);
+        var cut = RenderComponent<Pomodoro.Web.Pages.Index>();
+
+        await cut.Instance.HandleTimerStart();
```
[the remark this answers is not in the document]
Comment (Nayeem170) [reply tests/Pomodoro.Web.Tests/Pages/IndexCoverageTests.cs:175]: Fixed in PR #67 (commit 49b9751) - changed to \cut.InvokeAsync(() => cut.Instance.HandleTimerStart())\. Already merged into develop.

File: tests/Pomodoro.Web.Tests/Services/TaskServiceTests/TaskServiceTests.Repeat.cs [hunk 2]
```diff
@@ -131,6 +131,31 @@ public async Task CompleteTaskAsync_RecurringExpiredEnd_DoesNotSetNextOccurrence
...
+        MockIndexedDb.Setup(d => d.GetAsync<AppStateRecord>(It.IsAny<string>(), It.IsAny<string>()))
+            .ReturnsAsync((AppStateRecord?)null);
+
+        var service = CreateService();
+        await service.InitializeAsync();
+        await service.CompleteTaskAsync(taskId);
+
+        Assert.True(service.AllTasks[0].IsCompleted);
+        Assert.Null(service.AllTasks[0].Repeat.NextOccurrence);
+    }
```
[the remark this answers is not in the document]
Comment (Nayeem170) [reply tests/Pomodoro.Web.Tests/Services/TaskServiceTests/TaskServiceTests.Repeat.cs:158]: Already fixed in commit 09ff368 - the test captures \ar today = DateTime.UtcNow.Date\ at line 137 and uses it for both EndDate and LastCompletedDate.

File: codecov.yml [hunk 3]
```diff
@@ -5,11 +5,11 @@ coverage:
   status:
     project:
       default:
-        target: 98%
+        target: 99%
```
[the remark this answers is not in the document]
Comment (Nayeem170) [reply codecov.yml]: Fixed in commit 09ff368 - updated both project and patch targets to 99.5%.

File: tests/Pomodoro.Web.Tests/Components/History/WeeklyMiniChartTests.cs [hunk 4]
```diff
@@ -627,6 +627,34 @@ public Task WeeklyMiniChart_OnParametersSetAsync_DetectsChangedValues()
         return Task.CompletedTask;
     }
 
+    [Fact]
+    public Task WeeklyMiniChart_OnParametersSetAsync_SameData_SkipsReRender()
+    {
+        var jsRuntimeMock = new Mock<IJSRuntime>();
+        Services.AddSingleton<IJSRuntime>(jsRuntimeMock.Object);
+        Services.AddSingleton<IChartService>(new ChartService(jsRuntimeMock.Object));
+        Services.AddSingleton(new ChartDataFormatter());
```
[the remark this answers is not in the document]
Comment (Nayeem170) [reply tests/Pomodoro.Web.Tests/Components/History/WeeklyMiniChartTests.cs]: Fixed in commit 09ff368 - replaced Mock<IJSRuntime> with TestJsRuntime test double.

File: tests/Pomodoro.Web.Tests/Components/TaskItemComponentTests.cs [hunk 5]
```diff
@@ -569,5 +570,26 @@ public async Task TaskItemComponent_HandleKeyDown_Enter_WhenCompleted_DoesNotInv
...
+            IsCompleted = false,
+            Repeat = new RepeatRule { Type = RepeatType.Daily }
+        };
+
+        var cut = RenderComponent<TaskItemComponent>(parameters =>
+            parameters.Add(p => p.Item, task));
+
+        var badge = cut.Find(".task-badge");
+        badge.GetAttribute("title").Should().Be("Daily");
+        badge.ClassList.Should().NotContain("paused");
```
[the remark this answers is not in the document]
Comment (Nayeem170) [reply tests/Pomodoro.Web.Tests/Components/TaskItemComponentTests.cs]: Fixed in commit 09ff368 - changed assertion to check for repeat-paused instead of paused.

File: tests/Pomodoro.Web.Tests/Models/TaskItemTests/TaskItemTests.Repeat.cs [hunk 6]
```diff
@@ -0,0 +1,107 @@
+using FluentAssertions;
+using Pomodoro.Web.Models;
+using Xunit;
+
+namespace Pomodoro.Web.Tests.Models;
+
+[Trait("Category", "Model")]
+public class TaskItemRepeatTests
```
[the remark this answers is not in the document]
Comment (Nayeem170) [reply tests/Pomodoro.Web.Tests/Models/TaskItemTests/TaskItemTests.Repeat.cs]: Fixed in commit 09ff368 - changed [Trait(Category, Model)] to [Trait(Category, Service)].

File: tests/Pomodoro.Web.Tests/Pages/HistoryCoverageTests.cs [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+        var cut = RenderComponent<Pomodoro.Web.Pages.History>(parameters =>
+            parameters.Add(p => p.InitialSelectedDate, new DateTime(2023, 1, 5)));
+
+        var method = typeof(HistoryBase).GetMethod("UpdateFormattedDate", BindingFlags.Instance | BindingFlags.NonPublic);
+        method!.Invoke(cut.Instance, null);
+
+        var formattedProp = typeof(HistoryBase).GetProperty("FormattedSelectedDate", BindingFlags.Instance | BindingFlags.NonPublic);
+        var todayProp = typeof(HistoryBase).GetProperty("IsSelectedDateToday", BindingFlags.Instance | BindingFlags.NonPublic);
+        formattedProp!.GetValue(cut.Instance).Should().Be("Jan 5");
+        todayProp!.GetValue(cut.Instance).Should().Be(false);
```
[the remark this answers is not in the document]
Comment (Nayeem170) [reply tests/Pomodoro.Web.Tests/Pages/HistoryCoverageTests.cs]: Fixed in commit 09ff368 - replaced hardcoded Jan 5 with new DateTime(2023, 1, 5).ToString(MMM d).
